lldb/tools/lldb-dap/Handler/RequestHandler.h

```diff
@@ -57,235 +77,306 @@ class RequestHandler {
   DAP &dap;
 };
 
-class AttachRequestHandler : public RequestHandler {
-public:
-  using RequestHandler::RequestHandler;
+/// Base class for handling DAP requests. Handlers should declare their
+/// arguments and response body types like:
+///
+/// class MyRequestHandler : public RequestHandler<Arguments, ResponseBody> {
+///   ....
+/// };
+template <typename Args, typename Body>
+class RequestHandler : public BaseRequestHandler {
+  using BaseRequestHandler::BaseRequestHandler;
+
+  void operator()(const llvm::json::Object &request) const override {
+    /* no-op, the other overload handles json coding. */
+  }
```

Comment:
I think this wouldn't be necessary if you had something like a `LegacyRequestHandler` class which would declare the (abstract) function which takes the json object, and implements the protocol::Request version to forward to that). The base interface would only contain the protocol::Request overload.